pkg/manager/controllers/addonconfiguration/addon_configuration_reconciler.go

```diff
@@ -72,15 +75,50 @@ func (d *managedClusterAddonConfigurationReconciler) reconcile(
 		errs = append(errs, err)
 	}
 
-	for _, addon := range graph.addonToUpdate() {
+	nonplacementaddons, placementaddons := graph.addonToUpdateByPlacement()
+	// update all non placement addons
+	if err := d.updateAddonsConfig(ctx, nonplacementaddons); err != nil {
+		errs = append(errs, err...)
+	}
+
+	// update placement addons
+	for _, placementStrategy := range cma.Spec.InstallStrategy.Placements {
+		rollouttype := placementStrategy.RolloutStrategy.Type
+		switch {
+		case rollouttype == "" || rollouttype == addonv1alpha1.AddonRolloutStrategyUpdateAll:
+			if addons, exist := placementaddons[placementStrategy.PlacementRef]; exist {
+				if err := d.updateAddonsConfig(ctx, addons); err != nil {
+					errs = append(errs, err...)
+				}
+			}
+		case rollouttype == addonv1alpha1.AddonRolloutStrategyRollingUpdate:
+			placementNodes := graph.getPlacementNodes()
+			if installStrategyNode, exist := placementNodes[placementStrategy.PlacementRef]; exist {
+				length, err := parseMaxConcurrency(placementStrategy.RolloutStrategy.RollingUpdate.MaxConcurrency, len(installStrategyNode.clusters))
```

Comment:
again I think this could be all done when building graph

Reply:
fixed.